docs/getting-started/kubernetes/crds.mdx

```diff
@@ -0,0 +1,200 @@
+
+## 1. Install the ngrok Kubernetes Operator
+
+<K8sSetup />
+<K8sInstall />
+
+## 2. Get your domain
+
+Head over to the [**Domains** section](https://dashboard.ngrok.com/domains) and click **+ New Domain** to grab a new domain.
+You can choose between a static domain with an ngrok-branded TLD like `.ngrok.app` or `.ngrok.io`, or you can [bring a custom domain](/docs/universal-gateway/custom-domains/).
+
+We'll refer to this as `$YOUR_DOMAIN` from here on out.
+
+## 3. Create your endpoint manifest
+
+You have two options to expose your Kubernetes service with ngrok's CRDs:
+
+1. A single `AgentEndpoint` resource: This option is simplest, but is typically limited to exposing one service on a hostname.
+2. A pair of `CloudEndpoint` and `AgentEndpoint` resources: This option allows you to route traffic to any number of Kubernetes services from a single hostname depending on their path, headers, and more using [expressions](/docs/traffic-policy/concepts/expressions/) and the [`forward-internal` Traffic Policy action](/docs/traffic-policy/actions/forward-internal).
+
+This manifest exposes a Kubernetes Service named `hello-world` listening on port `80` on `$YOUR_DOMAIN`.
+Change these three values according to your specific deployment.
+
+<Tabs groupId="crds" queryString="crd-option">
+  <TabItem value="agent-endpoint" label="AgentEndpoint" default>
+
+```yaml title="ngrok-crds.yaml"
+apiVersion: ngrok.k8s.ngrok.com/v1alpha1
+kind: AgentEndpoint
+metadata:
+  name: hello-world-endpoint
+spec:
+  url: https://$YOUR_DOMAIN
+  upstream:
+    url: http://hello-world.default:80
+```
+
+  </TabItem>
+  <TabItem value="cloud-agent-endpoint" label="CloudEndpoint + AgentEndpoint">
+
+```yaml title="ngrok-crds.yaml"
+apiVersion: ngrok.k8s.ngrok.com/v1alpha1
+kind: CloudEndpoint
+metadata:
+  name: hello-world-cloud-endpoint
+spec:
+  url: $YOUR_DOMAIN
+trafficPolicy:
+  policy:
+    on_http_request:
+      - actions:
+        - type: forward-internal
+          config:
+            url: http://hello-world.internal
+---
+apiVersion: ngrok.k8s.ngrok.com/v1alpha1
+kind: AgentEndpoint
+metadata:
+  name: hello-world-endpoint
+spec:
+  url: http://hello-world.internal
+  upstream:
+    url: http://hello-world.default:80
+```
+
+  </TabItem>
+</Tabs>
+
+Apply the manifest with `kubectl apply -f ngrok-crds.yaml`.
+
+The ngrok Kubernetes Operator watches your cluster for `AgentEndpoint` and `CloudEndpoint` resources you create and provisions a new agent endpoint to forward traffic into your Service.
+
+You can now send a request to `$NGROK_DOMAIN` with your browser or `curl`.
+Your Kubernetes service is now available to anyone on the public internet!
+
+## 4. Secure your app with Traffic Policy
+
+In cases where you need to restrict access to your API or app, ngrok's [Traffic Policy engine](/docs/traffic-policy) and the [`restrict-ips`](/docs/traffic-policy/actions/restrict-ips) let you quickly allow only certain addresses to access your endpoint.
```

Comment:
```suggestion
In cases where you need to restrict access to your API or app, ngrok's [`restrict-ips`](/docs/traffic-policy/actions/restrict-ips) [Traffic Policy](/docs/traffic-policy) action lets you quickly allow only certain addresses to access your endpoint.
```

I _think_ we don't commonly refer to it as an engine in the docs. I think we generally refer to it as a common expression language. So I thought I'd cut reference to that as it might potentially confuse people.